TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php

```diff
@@ -461,17 +461,17 @@ protected function replaceNodeData(NodeInterface $node, NodeData $targetNodeData
             }
         }
 
-        if ($node->isRemoved() === true) {
-            $this->nodeDataRepository->remove($targetNodeData);
-        } else {
-            $targetNodeData->similarize($node->getNodeData());
-            if ($nodeWasMoved) {
-                $targetNodeData->setPath($node->getPath(), false);
-            }
-            $targetNodeData->setLastPublicationDateTime($this->now);
-            $node->setNodeData($targetNodeData);
-            $this->nodeService->cleanUpProperties($node);
+        $targetNodeData->setRemoved($node->isRemoved());
+        $targetNodeData->similarize($sourceNodeData);
+        // TODO: This seemswrong and introduces a publish order between nodes. We should always set the path.
```

Comment:
Well, it is a behavior change and there is a test covering this (but that behavior change was introduced only in 2.1) so I didn't want to decide that in this change...